### DISCUSSION TITLE
Select the same screen region automatically?

### DISCUSSION THREAD

**Discussion by Twixtfanatic** [flameshot-org/flameshot#3181] [Q&A] [answered]
Is there a way to select the same region automatically, without having to define the region with the mouse again and again? My OS is Linux Mint xfce. Thanks

**Comment by mmahmoudian** [comment, marked as answer]
Yes it is possible. You can do for example the following to select a 400 by 300 rectangle at x=100 and y=200

```sh
flameshot gui --region 400x300+100+200
```

You can read more about CLI here:

https://flameshot.org/docs/advanced/commandline-options/

**Reply by Twixtfanatic** [reply]
That's great news, and I appreciate the info, but I still need a way to determine what the numbers are for my desired screenshot. I have to use some other app for this, or trial and error. (I'm not asking for your help to do this. I'm just trying to make a point here.) Then after that, I have to type the command line into a terminal. Compare this with many other screen capture utilities which provide a keyboard or mouse shortcut to repeat the most recent capture. I really like the built-in graphics editor and command line flexibility that Flameshot provides. But I can well imagine that other whiny users like me would stop using Flameshot for this reason alone. IMO that would be a shame. Perhaps I should post in the suggestions forum? Thanks for taking the time.

And while I'm whining, I notice the save-to-file feature is sometimes a bit flaky for me. I click in the text entry field to name the file and am unable to type anything there. I keep clicking and eventually I am able to enter text. Is this a known bug? Thanks again.

**Reply by mmahmoudian** [reply]
If you read the link I previously provided, you will see that there is another CLI argument (`--print-geometry`) that prints the dimension and size of the region you select. This means you don't need to find the correct numbers by try and error.

I demonstrated before to another user on how to use these in combination:

https://github.com/flameshot-org/flameshot/issues/1509#issuecomment-965103716

And

https://github.com/flameshot-org/flameshot/issues/1509#issuecomment-1117332447

But if you want to always use the last region, you can enable that from the Flameshot configurations to always use the last region. If you can't find it, message here and I will guide you.

**Reply by alturic** [reply]
Not to necro this, but either the portable version doesn't support this, or this option broke sometime between 5/2/2023 and today. I have the option checked in the gui but when I hit print screen it never shows the last region captured?

**Reply by mmahmoudian** [reply]
@alturic portable version in which operating system? Also, can you share your Flameshot config file.

Can you also make sure that there is no other instances of Flameshot installed on that machine.

**Reply by alturic** [reply]
> @alturic portable version in which operating system? Also, can you share your Flameshot config file.
> 
> Can you also make sure that there is no other instances of Flameshot installed on that machine.

Windows 10.

There is no other instances installed as I'm using the portable version. This is the only instance running of course.

[flameshot.txt](https://github.com/user-attachments/files/16266319/flameshot.txt)

**Reply by mmahmoudian** [reply]
@alturic I don't have access to Windows machine to test this, but I have a suggestion: try the portable version from our development builds (this would include the latest changes in the code) and see if it still behaves the same: https://flameshot.org/docs/installation/development-build/

**Edit:** Somehow the nightly.link website is not working properly. you can go this this next link, scroll to the bottom of the page, and download the portable version according to your CPU:
https://github.com/flameshot-org/flameshot/actions/runs/9761265614#artifacts

**Reply by alturic** [reply]
> @alturic I don't have access to Windows machine to test this, but I have a suggestion: try the portable version from our development builds (this would include the latest changes in the code) and see if it still behaves the same: https://flameshot.org/docs/installation/development-build/
> 
> **Edit:** Somehow the nightly.link website is not working properly. you can go this this next link, scroll to the bottom of the page, and download the portable version according to your CPU: https://github.com/flameshot-org/flameshot/actions/runs/9761265614#artifacts

Alright, so this version looks to have a working last region, however with a multi-monitor setup it's putting the region on a different monitor than the one the screenshot was previously taken on. I wouldn't even be against that (technically, but wouldn't be ideal) but it's not even putting the region on the "primary" monitor which was the monitor the previous screenshot was taken from.

If I could make a feedback type of comment, what would be utterly fantastic would be the last region to work (of course, duh heh), but if it could even remember the monitor the last screenshot was taken on. That would be freaking amazing. :P

**Edit:** Interesting tidbit after a little more testing, if I take/took a screenshot from one of the other 3 monitors, the last region never works. So the only way I can get it to work (including on the actual release version 12.1 which I just tested to confirm), is to take the screenshot on the primary monitor only. However, everything with my comment above still remains true even on 12.1 as well. If I take the screenshot on the primary monitor (a windows setting/naming convention) the last region doesn't ever seem to get put on that monitor again *and* if I take a screenshot on any of the secondary monitors, use last region doesn't work at all.

**Reply by mmahmoudian** [reply]
One question: do you connect your external monitors after the computer has been turned on?

**Reply by alturic** [reply]
No.

**Reply by alturic** [reply]
Did you have any chance to look into this at all? :(
